packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1520,8 +1519,8 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
 
   @override
   Iterable<OverlayEntry> createOverlayEntries() sync* {
-    yield _modalBarrier = OverlayEntry(builder: _buildModalBarrier);
-    yield _modalScope = OverlayEntry(builder: _buildModalScope, maintainState: maintainState);
+    yield _modalBarrier = OverlayEntry(builder: _buildModalBarrier, onWidgetStatusChanged: lockDisposal);
```

Comment:
This seems to again only tie this mechanism to ModalRoutes?

Reply:
We need to document somewhere that OverlayEntries created in this method need to have their onWidgetStatusChanged wired up to lockDisposal.